File: 011.container-with-most-water/011.container-with-most-water.java
//方法1（Java版）：暴力法  
//在这种情况下，我们将简单地考虑每对可能出现的线段组合并找出这些情况之下的最大面积。

public class Solution {
    public int maxArea(int[] height) {
        int maxarea = 0;
        for (int i = 0; i < height.length; i++)
            for (int j = i + 1; j < height.length; j++)
                maxarea = Math.max(maxarea, Math.min(height[i], height[j]) * (j - i));
        return maxarea;
    }
}

File: 011.container-with-most-water/011.container-with-most-water.cpp
//方法4（leetcode版）：逼近法
//每次左右两端都舍去短的那一端 
//若选择短的那一端【S=小于等于此端的高*小于等于当前的最大区间长度】，至多的面积也是选择最左最右的一个矩形，因此不必再考虑短的一端，直接舍去逼近

class Solution {
public:
    int maxArea(vector<int>& height) {
    int _m = 0, r = height.size()-1,l=0;
    while (r > l) {
        _m = max(_m, min(height[l], height[r])*(r - l));
        if (height[r] > height[l]) ++l;
        else --r;
    }
    return _m;
    }
};
